src/overlay/Floodgate.cpp

```diff
@@ -120,19 +124,32 @@ Floodgate::broadcast(StellarMessage const& msg, bool force)
         releaseAssert(peer.second->isAuthenticated());
         if (peersTold.insert(peer.second->toString()).second)
         {
-            mSendFromBroadcast.Mark();
-            std::weak_ptr<Peer> weak(
-                std::static_pointer_cast<Peer>(peer.second));
-            mApp.postOnMainThread(
-                [smsg, weak, log = !broadcasted]() {
-                    auto strong = weak.lock();
-                    if (strong)
-                    {
-                        strong->sendMessage(smsg, log);
-                    }
-                },
-                fmt::format(FMT_STRING("broadcast to {}"),
-                            peer.second->toString()));
+            if (msg.type() == TRANSACTION && peer.second->shouldFloodLazily())
+            {
+                mMessagesAdvertized.Mark();
```

Comment:
I think it’s useful to know the number of tx hashes advertised. Knowing the number of hashes & the number of adverts would give us more insights than only the number of adverts. Would it make sense to keep track of it inside `queueTxHashToAdvertize`?

Reply:
I agree with Hide that it'd be useful to know how many hashes are advertised. That being said, it's true that mSendFromBroadcast tracks something similar, however for the time of the rollout, we probably want to separate them, otherwise we won't know which ones are pull mode and which ones are not. Once the rollout is complete, we can remove the other metric.